sklearn/cluster/_optics.py

```diff
@@ -603,15 +612,16 @@ def _set_reach_dist(
     # Only compute distances to unprocessed neighbors:
     if metric == "precomputed":
         dists = X[point_index, unproc]
+        if issparse(dists):
+            dists.sort_indices()
+            dists = dists.data
```

Comment:
@jeremiedbb This is a change from your previous approval. Are you okay with this?